export/kakfa_test.go

```diff
@@ -38,42 +59,191 @@ var (
 func TestNewKafka(t *testing.T) {
 	got, err := NewKafka(config, kafkaConfig, topicInfo)
 	if err != nil {
-		t.Fatalf("Error creating new Kafka")
+		t.Errorf("Error creating new Kafka: %v", err)
 	}
-	defer got.Stop()
+	got.Stop()
 
 	gotKafka := got.Exporter.(Kafka)
 	compareKafka(t, *kafkaExporter, gotKafka)
-
 }
 
 func TestSetMessageFlushTime(t *testing.T) {
 	got, err := NewKafka(config, kafkaConfig, topicInfo)
 	defer got.Stop()
 	if err != nil {
-		t.Errorf("Error creating new Kafka")
+		t.Errorf("Error creating new Kafka: %v", err)
 	}
 	got.SetMessageFlushTime(newFlushTime)
 
 	gotKafka := got.Exporter.(Kafka)
 	compareKafka(t, *kafkaExporterFlushTime, gotKafka)
+}
+
+func TestKafkaCreateTopic(t *testing.T) {
+	cli, network := createDockerNetwork(t, dockerNetwork)
+	defer removeDockerNetwork(t, cli, network)
+
+	zookeeper := startZookeeperContainer(t, zookeeperImage, zookeeperContainerName, dockerNetwork, zookeeperPort)
+	defer shutdownContainer(t, zookeeper)
+
+	kafkaServer := startKafkaContainer(t, kafkaImage, dockerNetwork, zookeeperContainerName, zookeeperPort, "false", kafkaPort)
+	defer shutdownContainer(t, kafkaServer)
+
+	kafkaConfig := getKafkaConfig(t, kafkaServer, kafkaPort)
+
+	if err := createTopic(topicCreationInfo, kafkaConfig); err != nil {
+		t.Errorf("Failed to create topic: %v", err)
+	}
+}
+
+func TestKafkaExportMetrics(t *testing.T) {
```

Comment:
Can we add a test case for the filter to make sure the actual metrics exported to kafka are the same as expected? We can use the kafka consumer example [here](https://github.com/confluentinc/confluent-kafka-go#examples) to consume the messages and compare the actual and expected metrics.

Reply:
captured in a JIRA https://confluentinc.atlassian.net/jira/software/projects/METRICS/boards/173?selectedIssue=METRICS-2172